test/integration/crl_test.go

```diff
@@ -97,3 +101,84 @@ func TestCRLPipeline(t *testing.T) {
 	test.AssertEquals(t, string(reason), "5")
 	resp.Body.Close()
 }
+
+func TestTemporalAndExplicitShardingCoexist(t *testing.T) {
+	db, err := sql.Open("mysql", vars.DBConnSAIntegrationFullPerms)
+	if err != nil {
+		t.Fatalf("sql.Open: %s", err)
+	}
+	// Insert an old, revoked certificate in the certificateStatus table. Importantly this
+	// serial has the 7f prefix, which is in test/config-next/crl-updater.json in the
+	// `temporallyShardedPrefixes` list.
+	oldSerial := "7faa39be44fc95f3d19befe3cb715848e601"
+	issuerID := 43104258997432926
+	_, err = db.Exec(`DELETE FROM certificateStatus WHERE serial = ?`, oldSerial)
+	if err != nil {
+		t.Fatalf("deleting old certificateStatus row: %s", err)
+	}
+	_, err = db.Exec(fmt.Sprintf(`
+		INSERT INTO certificateStatus (serial, issuerID, notAfter, status, ocspLastUpdated, revokedDate, revokedReason, lastExpirationNagSent)
+		VALUES ("%s", %d, "%s", "revoked", NOW(), NOW(), 0, 0);`,
+		oldSerial, issuerID, time.Now().Add(24*time.Hour).Format("2006-01-02 15:04:05")))
+	if err != nil {
+		t.Fatalf("inserting old certificateStatus row: %s", err)
+	}
+
+	client, err := makeClient()
```

Comment:
Comment saying something like `// Create and revoke a new, explicitly-sharded cert.`

Although this made me realize: this test name says that it is testing whether temporal and explicit sharding coexist. But in //test/config/ it's only testing whether temporal and temporal sharding coexist, since the modern cert will still be temporally sharded.

I think it's better to run this test in both envs, rather than gate it off, but possibly worth leaving a comment to this effect as well.